src/main/java/minijava/ir/optimize/ConstantFolder.java

```diff
@@ -177,12 +253,20 @@ public void visit(Or node) {
 
   @Override
   public void visit(Phi node) {
-    phiStrategy.visit(node);
+    TargetValue newValue =
+        seq(node.getPreds())
+            .map(this::getValue)
+            .reduce(TargetValue.getUnknown(), ConstantFolder::supremum);
+    updateValue(node, newValue);
```

Comment:
This is basically copied from the `SupremumStrategy`. I'm not sure what workaround the other strategy did, but I'm pretty confident this works.

I guess the other strategy tried to work around the fact that we also incorporated `Load`, `Store`, `Call`, `Cond`, etc. somehow. Now, these nodes are just assigned a `TargetValue.getBad()` (`defaultVisit`).